src/insights_client/run.py

```diff
@@ -3,8 +3,12 @@
 
 from insights import package_info
 
+import logging
 import metrics
 import utc
+from requests import ConnectionError
```

Comment:
Shouldn’t we keep _requests_ contained in the _metrics_ module? It’s a bit unfortunate that _metrics.MetricsHTTPClient_ is a _requests.Session_ and that it’s _post_ method does violate its parent’s signature, so we can’t really keep all the _requests_ stuff inside. But all _RequestExceptions_ including _ConnectionError_ [are](https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/exceptions.py#L12) _IOErrors_. So this should not be needed at all. See below.